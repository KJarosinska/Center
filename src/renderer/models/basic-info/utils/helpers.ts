import { SimCard } from "Renderer/models/basic-info/interfaces"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter(simCard => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
