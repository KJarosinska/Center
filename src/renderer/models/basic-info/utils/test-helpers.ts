import { IpcRequest } from "Common/requests/ipc-request.enum"

export const commonCalls = {
  [IpcRequest.GetDeviceInfo]: Promise.resolve({
    name: "Ziemniaczek",
    modelName: "U12300000",
    modelNumber: "A1239999",
    serilaNumber: "a-b-3d",
    osVersion: "0.123v",
    osUpdateDate: "12-12-2003",
  }),
  [IpcRequest.GetNetworkInfo]: Promise.resolve({
    simCards: [
      {
        active: true,
        carrier: "AAAAAAAAAAAA",
        iccid: 1234,
        imei: 5678,
        meid: 8765,
        network: "Y-Mobile",
        networkLevel: 0.5,
        number: 12345678,
        seid: "1234",
        slot: 1,
      },
      {
        active: false,
        carrier: "BBBBBBBBBBBB",
        iccid: 412,
        imei: 42,
        meid: 1410,
        network: "X-Mobile",
        networkLevel: 0.69,
        number: 7001234523,
        seid: "x123",
        slot: 2,
      },
    ],
  }),
  [IpcRequest.GetStorageInfo]: Promise.resolve({
    capacity: 9001,
    available: 99999999999999,
    categories: [
      { label: "music", filesCount: 1233333, size: 999999999 },
      { label: "storage", filesCount: 100000, size: 999999999 },
    ],
  }),
  [IpcRequest.GetBatteryInfo]: Promise.resolve({
    level: 9001,
    charging: false,
    maximumCapacity: 99999,
  }),
  [IpcRequest.GetBackupsInfo]: Promise.resolve({
    backups: [
      {
        createdAt: "20-11-15T07:35:01.562Z20",
        size: 99999,
      },
      {
        createdAt: "20-01-30T07:35:01.562Z20",
        size: 1234567,
      },
    ],
  }),
}
