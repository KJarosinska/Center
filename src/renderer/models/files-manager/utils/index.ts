import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
