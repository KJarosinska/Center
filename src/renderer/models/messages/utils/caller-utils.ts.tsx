import { Author } from "Renderer/models/messages/messages.interface"

const removeDecoratorsFromPhoneNumber = (string = ""): string => {
  return string.split(" ").join("").replace("+", "")
}

export interface CallerSearchParams extends Record<string, string> {
  id: string
  phoneNumber: string
}

export const isCallerMatchingSearchParams = (
  caller: Author,
  params: CallerSearchParams
): boolean => {
  return (
    params.id === caller.id &&
    removeDecoratorsFromPhoneNumber(caller.primaryPhoneNumber) ===
      removeDecoratorsFromPhoneNumber(params.phoneNumber)
  )
}
