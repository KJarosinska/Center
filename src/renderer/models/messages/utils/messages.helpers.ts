import { Contact } from "Renderer/models/phone/phone.typings"
import { Topic } from "Renderer/models/messages/messages.interface"

export type ContactsCollection = Record<string, Contact>
export type Getter = (
  id: string,
  collection: ContactsCollection
) => Contact | undefined

export const getContactsAsMap = (contacts: Contact[]) => {
  return contacts.reduce((acc: ContactsCollection, item: Contact) => {
    return {
      ...acc,
      [item.id]: item,
    }
  }, {})
}

export const getContactDetails = (
  id: string,
  collection: ContactsCollection
): Contact | undefined => {
  if (id in collection) {
    return collection[id]
  }

  return undefined
}

export const expandTopic = (
  topic: Topic,
  collection: ContactsCollection,
  getter: Getter
) => {
  const { messages } = topic

  return {
    ...topic,
    messages: messages.map((msg) => {
      const author = getter(msg.author.id, collection)

      if (author) {
        return {
          ...msg,
          author,
        }
      }

      return msg
    }),
  }
}

export const createFullMessagesCollection = (state: any) => {
  const {
    messages: { topics },
    phone: { contacts: baseContacts },
  } = state

  const contacts = getContactsAsMap(baseContacts)

  return topics.map((topic: Topic) => {
    const { id } = topic.caller
    const caller = getContactDetails(id, contacts)

    if (caller) {
      return {
        ...expandTopic(topic, contacts, getContactDetails),
        caller,
      }
    }

    return topic
  })
}
