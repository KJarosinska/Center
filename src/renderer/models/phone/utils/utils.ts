import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
