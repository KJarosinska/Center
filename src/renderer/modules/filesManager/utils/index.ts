import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
