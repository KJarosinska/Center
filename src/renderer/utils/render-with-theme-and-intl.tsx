import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"

const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult<Q>
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q> | Omit<RenderOptions, "queries">
) {
  return render<Q>(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={LANGUAGE.default}
        locale={LANGUAGE.default}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
