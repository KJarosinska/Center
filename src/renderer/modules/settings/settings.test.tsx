import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import "@testing-library/jest-dom/extend-expect"
import React from "react"
import SettingsUI from "Renderer/components/rest/settings/settings-ui.component"

test("matches snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <SettingsUI appAutostart={false} appTethering={false} />
  )
  expect(container).toMatchSnapshot()
})
