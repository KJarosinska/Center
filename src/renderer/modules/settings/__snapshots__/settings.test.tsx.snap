// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c2 {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.05;
  color: #6a6a6a;
  font-weight: 300;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
}

.c6 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
}

.c13 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c8 {
  grid-area: Actions;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 9rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c11:hover {
  background: #232426;
}

.c11 g {
  fill: #ffffff;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  border-radius: 0.5rem;
  border: 0.1rem solid #3b3f42;
}

.c15 g {
  fill: #000000;
}

.c15:hover {
  border-color: #000000;
}

.c14 {
  white-space: nowrap;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 {
  margin-right: 4rem;
}

.c12 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  z-index: 3;
  padding: 0 3.6rem;
  width: 75%;
}

.c12:disabled {
  pointer-events: none;
}

.c12:hover {
  z-index: 2;
}

.c12:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c12:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c12:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c16 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  padding: 0 3.6rem;
  width: 75%;
}

.c16:disabled {
  pointer-events: none;
}

.c16:hover {
  z-index: 2;
}

.c16:not(:hover) {
  border: 0.1rem solid #cdcecf;
  color: #cdcecf;
}

.c16:not(:hover) g {
  fill: #cdcecf;
}

.c16:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c16:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c16:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c4 {
  display: grid;
  grid-template-columns: 11rem 1fr 9rem;
  grid-template-areas: "Checkbox . Actions";
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 9rem;
  min-height: 9rem;
  box-sizing: border-box;
  border-bottom: solid 0.1rem #d2d6db;
  -webkit-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
  grid-template-areas: "Checkbox Actions";
  grid-template-columns: 1fr 15rem;
  border-bottom: solid 0.1rem #d2d6db;
}

.c4:hover {
  background-color: #f4f5f6;
}

.c4:hover .sc-fzoLsD {
  opacity: 1;
  visibility: visible;
}

.c5 {
  grid-area: Checkbox;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  grid-area: Name;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  margin-bottom: 0.4rem;
  margin-left: 4rem;
}

.c1 {
  border-bottom: solid 0.1rem #d2d6db;
}

.c3 {
  margin-left: 4rem;
  margin-bottom: 3.2rem;
}

.c0 {
  padding-top: 3.2rem;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2 c3"
      >
        [value] view.name.settings.description
      </p>
    </div>
    <div
      class="sc-fznKkj c4"
    >
      <div
        class="c5"
      >
        <p
          class="c6 sc-Axmtr c7"
        >
          [value] view.name.settings.autostartLabel
        </p>
      </div>
      <div
        class="c8"
      >
        <section
          class="c9 c10"
        >
          <button
            class="c11 sc-fzoiQi c12"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c13 c14"
            >
              [value] view.name.settings.offLabel
            </p>
          </button>
          <button
            class="c15 sc-fzoiQi c16"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c13 c14"
            >
              [value] view.name.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
    <div
      class="sc-fznKkj c4"
    >
      <div
        class="c5"
      >
        <p
          class="c6 sc-Axmtr c7"
        >
          [value] view.name.settings.tetheringLabel
        </p>
      </div>
      <div
        class="c8"
      >
        <section
          class="c9 c10"
        >
          <button
            class="c11 sc-fzoiQi c12"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c13 c14"
            >
              [value] view.name.settings.offLabel
            </p>
          </button>
          <button
            class="c15 sc-fzoiQi c16"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c13 c14"
            >
              [value] view.name.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
  </section>
</div>
`;
